provider/core/src/hello_world.rs

```diff
@@ -193,3 +193,18 @@ fn test_iter() {
         ]
     );
 }
+
+/// Marker type for [`Cow`]`<str>` where the backing cart is `str`.
+#[allow(clippy::exhaustive_structs)] // marker type
+pub struct CowStrMarker;
```

Comment:
Nit: It does not make sense to have this in `hello_world` because it is not used by `hello_world`. I disagree that this is a "toy" implementation; it could very well be used by data structs that contain only a string.

Reply:
It feels like one though. It's very easy for clients to write their own `CowStrMarker`, and do we actually want to provide a non-`ResourceMarker` `DataMarker` as a stable type? Also, why just `CowStrMarker` and not `CowBytesMarker`?

I'd also be happy with removing it.

Reply:
I won't be heartbroken if you remove it, but anywhere it's used, we should migrate to something else. It is mostly used in docs tests. You can change those to something else like HelloWorldV1 if you prefer, or by re-defining CowStrMarker at each call site. However, namespacing it in `hello_world` is not correct.

Reply:
Thought: CowStrMarker has convenience functions like `DataPayload::<CowStrMarker>::from_static_str`. You could achieve mostly the same goal by adding `DataPayload::<HelloWorldV1Marker>::from_static_str` or `HelloWorldV1::from_static_str("...").into_data_payload()`